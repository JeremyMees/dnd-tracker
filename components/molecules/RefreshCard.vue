<script setup lang="ts">
defineEmits<{ refresh: [] }>()
</script>

<template>
  <Card
    color="danger"
    class="w-full max-w-prose mx-auto flex flex-col sm:flex-row sm:items-center sm:justify-between gap-4 mb-6"
  >
    <h2 class="text-center">
      {{ $t('general.error.text') }}
    </h2>
    <button
      class="btn-foreground"
      :aria-label="$t('actions.tryAgain')"
      @click="$emit('refresh')"
    >
      {{ $t('actions.tryAgain') }}
    </button>
  </Card>
</template>
