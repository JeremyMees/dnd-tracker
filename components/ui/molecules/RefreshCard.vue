<script setup lang="ts">
defineEmits<{ refresh: [] }>()

const { t } = useI18n()
</script>

<template>
  <Card
    color="danger"
    class="w-full max-w-prose mx-auto flex flex-col sm:flex-row sm:items-center sm:justify-between gap-4 mb-6"
  >
    <h2 class="text-center">
      {{ t('general.error.text') }}
    </h2>
    <button
      class="btn-black"
      :aria-label="t('actions.tryAgain')"
      @click="$emit('refresh')"
    >
      {{ t('actions.tryAgain') }}
    </button>
  </Card>
</template>
