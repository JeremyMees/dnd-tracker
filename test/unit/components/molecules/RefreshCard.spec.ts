import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'
import RefreshCard from '~/components/molecules/RefreshCard.vue'

describe('RefreshCard', async () => {
  it('Should match snapshot', async () => {
    const component = await mountSuspended(RefreshCard)

    expect(component.html()).toMatchSnapshot()
  })

  it('Should be emit refresh event', async () => {
    const component = await mountSuspended(RefreshCard)
    const button = component.find('[data-test-refresh-button]')

    expect(button.exists()).toBeTruthy()

    await button.trigger('click')
    await nextTick()

    expect(component.emitted('refresh')).toBeTruthy()
  })
})
