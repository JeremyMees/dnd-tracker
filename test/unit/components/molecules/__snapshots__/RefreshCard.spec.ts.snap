// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RefreshCard > Should match snapshot 1`] = `
"<div class="transition-colors duration-300 border-4 rounded-lg p-4 border-destructive bg-destructive/50 w-full max-w-prose mx-auto flex flex-col sm:flex-row sm:items-center sm:justify-between gap-4 mb-6">
  <h2 class="text-center">general.error.text</h2><button data-test-refresh-button="" class="btn-foreground" aria-label="actions.tryAgain">actions.tryAgain</button>
</div>"
`;
